compiler/src/dmd/traits.d

```diff
@@ -739,6 +739,43 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
         auto se = new StringExp(e.loc, id.toString());
         return se.expressionSemantic(sc);
     }
+    if (e.ident == Id.fullyQualifiedName) // https://dlang.org/spec/traits.html#fullyQualifiedName
+    {
+        if (dim != 1)
+            return dimError(1);
+
+        Scope* sc2 = sc.push();
+        sc2.flags = sc.flags | SCOPE.noaccesscheck | SCOPE.ignoresymbolvisibility;
+        bool ok = TemplateInstance.semanticTiargs(e.loc, sc2, e.args, 1);
```

Comment:
This line is redundant. `TemplateInstance.semanticTiargs [is already called at the beginning of this function](https://github.com/dlang/dmd/pull/14711/files#diff-30bf71ba89939445e213ad60e223d2c317f12b698d71568fdab147e8693ab100L367).

Reply:
It's not quite the same. The `sc2.flags` are changed.

Reply:
Yes, but since semantic has already been performed all of the expressions have their type set and when they hit the expression semantic visit methods (which typically start with `if (exp.type) return`) no analysis is done. So, I suspect that this code only adds a couple of function calls that simply return.